lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -702,20 +709,85 @@ def _get_run_job_params_from_existing_job(job: Job, user_id: str) -> Dict:
         # Then the next fields are job inputs top level requirements, app run parameters, and scheduler resource requirements
         return run_job_params
 
+    def _check_ws_perms(
+        self, runjob_params, new_batch_job, batch_params, as_admin=False
+    ):
+        """
+        Check a single job, a single batch job, or a retry_multiple request with a mix of different jobs.
+        """
+        if as_admin:
+            return self.sdkmr.check_as_admin(requested_perm=JobPermissions.WRITE)
+        # Batch params have the next highest
+        if new_batch_job:
+            if batch_params:
+                return self._check_workspace_permissions(batch_params.get("wsid"))
+            # No ws perms to check
+            return
```

Comment:
Due to the if block structure this return isn't necessary

Reply:
Fixed

Reply:
:+1: